db/re/item_db.conf

```diff
@@ -151058,9 +151067,12 @@ item_db: (
 	Buy: 10
 	Loc: "EQP_SHADOW_SHIELD"
 	Script: <"
-		bonus2 bSubRace,RC_Insect,1;
-		if (getrefine()>=7) { bonus2 bSubRace,RC_Insect,1; }
-		if (getrefine()>=9) { bonus2 bSubRace,RC_Insect,1; }
+		if (getrefine() >= 9)
+			bonus2(bAddRaceTolerance, RC_Insect, 3);
+		if (getrefine() >= 7)
+			bonus2(bAddRaceTolerance, RC_Insect, 2);
```

Comment:
this line ... I think its wrong
if the equipment refine is 9, the value become 5 (3+2)

it should be ELSE IF ?

well logic wise, should use ELSE IF
but try check Aegis file, maybe it was really done like this officially

Reply:
You're right, Annie. It should be `else if`.
I'll fix this later today.

Reply:
```c
bonus2(bAddRaceTolerance, RC_Insect, 1 + (getrefine()>=7) + (getrefine()>=9));
```
hmm ??
just a suggestion

Reply:
Fixed.

> dunno if this kind of code is accepted

Nope, using a boolean expression in a calculation isn't valid.
But using a short-if would be valid:
```c
bonus2(bAddRaceTolerance, RC_Insect, 1 + ((getrefine() >= 7) ? 1 : 0) + ((getrefine() >= 9) ? 1 : 0));
```